### ISSUE TITLE
Editor window receives no mouse input in remote desktop connection with AnyDesk and TeamViewer

### ISSUE THREAD

**Issue by nekitu** [OurMachinery/themachinery-public#44]
Very low priority, but it was interesting to mention.
When remote desktop connection to a computer containing the editor running, the mouse input seem not to pass through to the editor window, but keyboard does.
This happens with AnyDesk and also with Teamviewer software.
I use remote when I go to a coffee shop with a mere tablet and I can use a keyboard to remote access a beefy computer at home running engine(s) at their full potential.
Again low prio but a peculiar behavior whatsoever, what are you using for input ? win32/xinput directly or sdl ?

**Comment by niklas-ourmachinery** [comment]
We are using [raw input](https://docs.microsoft.com/en-us/windows/win32/inputdev/raw-input). It might be that this doesn't work with these remote tools.

We probably won't change from using Raw Input as the default, since it gives more control, which is usually what you want for games. But we could maybe support regular mouse messages with a flag... or maybe even automatically switch to regular mouse messages if we don't detect a raw input mouse device...

But as you say, looking into this will probably be low prio for us.

**Comment by nekitu** [comment]
Yup, it was just an interesting issue, probably TV doesn't hook into the mouse input at that low level when replicating the events on the other side.

**Comment by pretty-wise** [comment]
I am experiencing the same issue using Microsoft Remote Desktop and Citrix on MacOS.

While searching about the possible solution to my problem I found a [post ](https://forum.unity.com/threads/problems-with-remote-desktop.806538/) with a comment: 

_"Turns out the problem actually was that we only responded to relative mouse input from RawInput whereas with RDP connections, you get absolute mouse input."_

I hope this helps.

**Comment by simon-ourmachinery** [comment]
Related report

https://forums.ourmachinery.com/t/unable-to-start-machinery-through-remote-desktop/1487

**Comment by frank-ourmachinery** [comment]
I added support for absolute positions from raw input. That seems to solve the issue with Microsoft's Remote Desktop, although rotating the viewport camera doesn't work as those are on relative events.

**Comment by niklas-ourmachinery** [comment]
I wonder if we have to generate our own relative events from the absolute ones for this case.

Though that might cause us to have "double" relative events... in the case where the relative events actually get sent.

**Comment by frank-ourmachinery** [comment]
From Discord channel (likely related):

> So I know this is a tiny tiny use case :slight_smile: so thought I'd pop it in here before creating a bug etc. I am using Sharemouse (www.sharemouse.com) to control a windows machine from my mac. I've done this for years when I need a windows machine for some part of a console or other build process but do 90%+ of my work on the mac. Unity/UE4 and most apps work perfectly well but Our Machinery does and odd thing. In the graph view (i.e. creation graph) the middle mouse button works to pan around the graph and the wheel zooms. In a 3D viewport, MMB or other mouse based navigation does not work, I get the crossed arrows cursor but the view does not move/rotate etc. (zoom with wheel does work). Also RMB + WASD does work.
So its a small(ish) thing and most likely not an Our Machinery problem, I only raise it as I not seeing the same problem with other apps.

https://discord.com/channels/690290013585408136/691973666140848148/844018102966288384
